iwyu_preprocessor.cc

```diff
@@ -645,6 +627,18 @@ void IwyuPreprocessorInfo::Ifndef(SourceLocation loc,
   FindAndReportMacroUse(GetName(id), id.getLocation());
 }
 
+// Clang will give a MacroExpands() callback for all macro-tokens
+// used inside an #if or #elif, *except* macro-tokens used within a
+// 'defined' operator. They produce a Defined() callback.
+void IwyuPreprocessorInfo::Defined(const Token& id,
+                                   const MacroDefinition& /*definition*/,
+                                   SourceRange /*range*/) {
+  ERRSYM(GetFileEntry(id.getLocation()))
+      << "[ #if defined ] " << PrintableLoc(id.getLocation())
```

Comment:
It seems the old method used a `SourceRange` instead of a `SourceLocation`. The Token type has `getEndLoc`/`getLastLoc`. Maybe you can use it to get a range like before?